crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs

```diff
@@ -152,14 +152,29 @@ pub(crate) fn verbose_decimal_constructor(checker: &mut Checker, call: &ast::Exp
             let Some(float) = float.as_string_literal_expr() else {
                 return;
             };
+
+            let normalized_float_string = float.value.to_str().trim().to_lowercase();
+
             if !matches!(
-                float.value.to_str().to_lowercase().as_str(),
-                "inf" | "-inf" | "infinity" | "-infinity" | "nan"
+                normalized_float_string.as_str(),
```

Comment:
```suggestion
                &normalized_float_string,
```

Reply:
actually had to revert that - type mismatch (`matches` doesn't let you compare `&String` and `&str`)

Reply:
I'm personally fine with `.as_str()`, but in case it's helpful @dylwil3, the way to make this compile would be to do this:

```suggestion
                &*normalized_float_string,
```

What that's saying to Rust is:
1. First dereference the `normalized_float_string` variable (the `*` dereferences things), converting the type from `String` to `str`. (A `String` is a smart pointer, where the pointer points to some underlying data of type `str`).
2. And then borrow it again, converting the type from `str` to `&str`

Something that comes up a lot in Ruff's linter rules is situations where you have a `&Box<Expr>`, but you need an `&Expr` to be able to match on it. One way to get from a `&Box<Expr>` to an `&Expr` is to use `as_ref()`, because there's this convenient implementation for `Box` in the Rust standard library that will take care of the conversion for you: https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-AsRef%3CT%3E-for-Box%3CT,+A%3E. But another way of doing that is to use the invocation `&**var` (where `var` has type `&Box<Expr>`): what that does is:
- Dereference `var` once, converting the type from `&Box<Expr>` to `Box<Expr>`.
- Dereference `var` _again_, converting the type from `Box<Expr>` to `Expr`. (`Box` is a smart pointer, just like `String`, so dereferencing the variable gets you from the pointer to the underlying data that the pointer is pointing to.)
- And then borrow `var`, converting the type from `Expr` to `&Expr`.

The `&*` and `&**` stuff can get some getting used to, but some people prefer it as it's a bit more explicit than some alternative idioms. I think that's more true with the `.as_ref()` situation than the `.as_str()` situation, though -- a `.as_ref()` call could really return anything (you have to lookup the implementation to know exactly), whereas I think it's fairly obvious from an `as_str` call what it's returning.

(Sorry if you already know some of this, just thought it might be helpful! It took me a while to figure some of this out.)